shop/src/components/payment-widget.js

```diff
@@ -19,16 +19,35 @@ const CONFIRMED_ROUND = 'confirmed-round';
  * https://github.com/PureStake/algosigner/issues/252 is resolved.
  * Task - https://www.pivotaltracker.com/story/show/178760753
  */
-async function getDefaultAccountAddr() {
-  const walletAccounts =
-    (await AlgoSigner.accounts({
-      ledger: CHAIN_NAME,
-    })) ?? [];
-
-  if (walletAccounts.length) {
-    return walletAccounts[0].address;
+async function getDefaultAccountAddr (selectedWallet, response) {
```

Comment:
Try to use break and return value from outside the switch case block. There is code duplication here in each switch case.

Reply:
A quick question, if I'm using return then I don't have to use break right? 
I don't understand what you mean by code duplication .? Is it this code used in two case ?
```
if (walletAccounts.length) {
        return walletAccounts[0].address;
      }
      return undefined;
    }
```

Reply:
Yes you don't need break when returning. The above code can moved outside switch case block. Something like this
switch (selectedWallet) {
    case Wallets.ALGOSIGNER: {
      walletAccounts =
        (await AlgoSigner.accounts({
          ledger: CHAIN_NAME,
        })) ?? [];
       break;
    }
 if (walletAccounts.length) {
        return walletAccounts[0].address;
      }
      return undefined;